erfanjazebnikoo/BenjaminFlutterCodingChallenge
# Feature: Complete 'Auto Complete' functionality for cities Text Field in Form

This Pull Request includes the initial stages of implementing the Auto Complete feature for cities, covering the following steps:

-**Autocomplete**: *Improve CustomTextFormField widget to support autocomplete options. Use this functionality in the first street address field. The widget should display list of hardcoded address suggestions when user has typed 2 or more characters. When field contains less than 2 characters, the field should not display any suggestions.*
-**API integration**: *Simulate real world scenario where addresses are fetched from external API*

### Testing Instructions:
1. Type a US city name in the Auto Complete field and verify suggested results. (example:  'Dallas')

These changes aim to enhance the user experience by adding Auto Complete functionality and improving quick access to city data.
